frontend/packages/dsco/src/checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,89 @@
+import classnames from 'classnames';
+import {useRef, useEffect, ChangeEvent, HTMLAttributes} from 'react';
+
+import {componentSizeToBodyTextSizeMap} from '@/common/constants';
+import {ComponentSizeXSToL} from '@/common/types';
+import Typography from '@/typography';
+
+import moduleStyles from './checkbox.module.scss';
+
+export interface CheckboxProps extends HTMLAttributes<HTMLInputElement> {
+  /** Checkbox checked state */
+  checked: boolean;
+  /** Checkbox onChange handler*/
+  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
+  /** The name attribute specifies the name of an input element.
+     The name attribute is used to reference elements in a JavaScript,
+     or to reference form data after a form is submitted.
+     Note: Only form elements with a name attribute will have their values passed when submitting a form. */
+  name: string;
+  /** The value attribute specifies the value of an input element. */
+  value?: string;
+  /** Checkbox label */
+  label?: string;
+  /** Is checkbox disabled */
+  disabled?: boolean;
+  /** Is checkbox indeterminate */
+  indeterminate?: boolean;
+  /** Size of checkbox */
+  size?: ComponentSizeXSToL;
```

Comment:
The only issue I can see is that the size selection doesn't match:
<img width="1032" alt="local" src="https://github.com/user-attachments/assets/88b58648-ccab-41fd-8f22-691cc6f99510" />

<img width="1030" alt="prod" src="https://github.com/user-attachments/assets/c2871935-f1ef-45ea-92f1-13d6ee278100" />

Reply:
Thank you for spotting this!

I've created a separate ticket for addressing this issue https://codedotorg.atlassian.net/browse/DESIGN2-253 as it's a common issue across all dsco package with `componentSizeXSToL`  and related to import strings, and plan to work on it separately in scope for all of the components once all the components will be moved over to dsco package.

If this is the only issue with this component - I suggest we go on and merge this pr.